test/framework/deployments/externalservice/kubernetes.go

```diff
@@ -60,29 +60,18 @@ func (k *k8SDeployment) Deploy(cluster framework.Cluster) error {
 }
 
 func (k *k8SDeployment) Delete(cluster framework.Cluster) error {
-	err := k8s.KubectlDeleteE(cluster.GetTesting(),
+	_ = k8s.KubectlDeleteE(cluster.GetTesting(),
 		cluster.GetKubectlOptions(externalServiceNamespace),
 		filepath.Join("testdata", fmt.Sprintf("%s.yaml", k.Name())))
-	if err != nil {
-		return err
-	}
 
 	framework.WaitPodsNotAvailable(externalServiceNamespace, k.Name())
 
-	pods, err := k8s.ListPodsE(cluster.GetTesting(), cluster.GetKubectlOptions(externalServiceNamespace), metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
+	// forcefully delete the namespace, no matter if there are other pods
```

Comment:
well, technically it matters, a bit. If I have 2 external services deployments and delete one of them, actually two of them will be wiped out.

What was the problem here?

Reply:
Indeed it matters, see the upper explanation for details. Essentially we want to ensure that all ExternalServices are not available before we start the next test. Unfortunatelly I did not find a reliable way to ensure the pods are not there anymore, instead, the proposal is to get rid of the namespace.